lib/yao/faraday_middlewares.rb

```diff
@@ -30,6 +31,39 @@ def call(env)
 end
 Faraday::Request.register_middleware os_token: -> { Faraday::Request::OSToken }
 
+class Faraday::Request::ReadOnly
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    return @app.call(env) if allowed_request?(env)
+
+    if Yao.config.raise_on_write
+      raise Yao::ReadOnlyViolationError
+    elsif Yao.config.noop_on_write
+      env
+    else
+      @app.call(env)
+    end
+  end
+
+  private
+
+  def allowed_request?(env)
+    return true if env[:method] == :get
+
+    allowed_requests = [
+      {method: :post, path: "/v2.0/tokens"}
+    ]
```

Comment:
Can it be a constant?

(Maybe every time `allowed_request?` is called, hash object is generated)